### DISCUSSION TITLE
/hyndsight/forecast-combinations/

### DISCUSSION THREAD

**Discussion by robjhyndman** [robjhyndman/robjhyndman.com#260] [Announcements]
R packages for forecast combinations 
 https://robjhyndman.com/hyndsight/forecast-combinations/

**Comment by robjhyndman** [comment]
```header
Originally posted by savvas on 2016-10-20 13:39:54
````
hi, i am a beginner as far as R is concerned and maybe what i ask is
simple but i can\'t reach it. Anyway here goes:How can i find confidence
intervals for the mixture in the opera example

**Comment by robjhyndman** [comment]
```header
Originally posted by savvas on 2016-10-21 13:51:05
````
hi, i wonder if you couldhelp me with fit1 \&lt;- hybridModel(myseries,
weights=&quot;equal&quot;). I tried to use a myseries as a timerseries(1x36) and
as a matrix(3x12).The first does not detect seasonality whereas the
second returns Error in hybridModel(group1matrix, weights = &quot;equal&quot;) :
The time series must be numeric and may not be a matrix or dataframe
object..The group1matrix is just like train without labels.why does it
raise an error?

**Comment by robjhyndman** [comment]
```header
Originally posted by Jason on 2016-11-29 06:53:07
````
He\'s simply giving some credit to packages that utilize his work.
Please don\'t use comments for help. He didn\'t write the packages.

**Comment by robjhyndman** [comment]
```header
Originally posted by Karthi Aru on 2017-01-20 15:22:58
````
sqrt is missing in the rmse calculation.

**Reply by robjhyndman** [reply]
```header
Originally posted on 2017-01-21 00:28:14
````
Thanks. Now fixed.

**Comment by robjhyndman** [comment]
```header
Originally posted by AnscombesGimlet on 2017-05-03 20:27:35
````
Hi Professor, I\'m not sure if you meant to put all of these warnings in
the blog, perhaps this is due to the conversion?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2017-05-03 21:02:17
````
Thanks. Now fixed.

**Comment by robjhyndman** [comment]
```header
Originally posted by Ben on 2017-06-05 14:59:19
````
Hi Professor, I wonder how to use Opera package for future predictions.
What I mean is how can I build real out-of-sample forecasts since do not
know the newY, because that\'s what I am actually trying to forecast. In
this case, I can not rain my weight.

What I currently did, I used latest 120 data points as training data for
weights, then use that best weights to forecast the future data. Also,
has a problems to calculate the prediction interval for opera model.
Thanks!

**Reply by robjhyndman** [reply]
```header
Originally posted by pankaj on 2017-09-14 09:50:04
````
I have the same question. Did you find some solution?\
Can we simply take the weights corresponding to the end of test set?

**Reply by robjhyndman** [reply]
```header
Originally posted by Vishal Gupta on 2017-10-09 08:25:06
````
I am stuck at the same problem, what approach have you used ?

**Reply by robjhyndman** [reply]
```header
Originally posted by pankaj on 2017-10-09 08:31:21
````
Didn\'t find a viable solution. So using simple average of forecasts. as
defined in
[https://cran.r-project.org/\...](https://cran.r-project.org/web/packages/forecastHybrid/forecastHybrid.pdf &quot;https://cran.r-project.org/web/packages/forecastHybrid/forecastHybrid.pdf&quot;)

**Reply by robjhyndman** [reply]
```header
Originally posted by Vishal Gupta on 2017-10-11 14:20:47
````
I got this from github, this implementation of opera gives weights for
real out-of-sample (i think)\
Take a look, what do you think !\
[https://github.com/Dralliag\...](https://github.com/Dralliag/opera)\
MLpol \&lt;- MLpol0\
for (i in 1:length(Y)) {\
MLpol \&lt;- predict(MLpol, newexperts = X\[i, \], newY = Y\[i\])\
}

**Reply by robjhyndman** [reply]
```header
Originally posted by Mert on 2017-12-27 03:23:16
````
Same problem here. Adjusting weight based on newY is a kind of
&quot;cheating&quot;, because we are trying to predict that newY.

My suggestion is let\'s assume we are going to predict 24 step ahead.
Firstly, predict 48 step ahead and use the first 24 step to find weights
and next 24 step to adjust forecasts with weight produced in the first
24 step. Do you think it is a valid idea?

Also, I am wondering this method is different than combining or ensemble
forecasts?

**Reply by robjhyndman** [reply]
```header
Originally posted by Burak on 2018-03-08 07:24:12
````
look at the output of mlpol, you will see that \$coefficients delivers
you the new weights for y+1

**Reply by robjhyndman** [reply]
```header
Originally posted by Yaman Skheita on 2018-04-10 08:15:24
````
One way to obtain future predictions is to run Opera and get the last
weights returned by the model.\
Run your different forecast methods and obtain future predictions.\
Multiply the prediction of each forecast method with the weight returned
by Opera.\
Try this approach and check the results.\
This is because Opera is giving weights (ratios) to different
forecasting methods based on their prediction accuracy compared to the
test sample.\
Check the github page with its example:
[https://github.com/dralliag\...](https://github.com/dralliag/opera/)

**Comment by robjhyndman** [comment]
```header
Originally posted by maddy on 2018-09-19 07:16:59
````
Hello Sir,\
I am little confused as I am getting different forecast values while
running same hybridmodel (insample one ) on same set of data .Can you
please guide why its is happening and how to get correct output using
this model . Using this model for weekly forecast for 4 weeks .

**Reply by robjhyndman** [reply]
```header
Originally posted on 2018-09-19 07:21:04
````
Ask on
[crossvalidated.com](http://crossvalidated.com &quot;crossvalidated.com&quot;)
or
[stackoverflow.com](http://stackoverflow.com &quot;stackoverflow.com&quot;).
I am not a one-man help service.

**Comment by ashley2876** [comment]
Hello! Thank you for all of the great info you post to your blog. I tried to Google the answer to my question, but have been unsuccessful. Is the hybrid model produced by forecastHybrid a true hybrid model, or rather a more simple ensemble model?

**Reply by robjhyndman** [reply]
It's an ensemble -- a weighted combination of several models. A "hybrid" approach can mean several things. e.g., it could mean that the residuals from some model are modelled with a second model, and sometimes this is iterated for additional models. In that sense, it is a form of boosting. Another possibility is that the components of one model are inputs into a second model. e.g., ETS states are input into a neural network.

**Reply by ashley2876** [reply]
Thank you so much, I appreciate it!
